### DISCUSSION TITLE
Time abstraction in .NET 8

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#63] [General]
# Time abstraction in .NET 8

With the upcoming release of .NET, the team introduced an abstraction of time itself. That can bring you major benefits especially if you have to test scenarios where time is a crucial part! Until now, you had to create your own wrapper. This, of course, makes integration with 3rd party libraries tricky.

https://steven-giesel.com/blogPost/4d5b70fe-35c1-43c7-bebe-49194195ffcb

**Comment by ghadzhigeorgiev** [comment]
Kind of an obvious thing to say, but... FINALLY! :)

**Reply by linkdotnet** [reply]
Exactly :) took long enough. Really helps making some tests easier. Especially with Task.Delay and friends to simplify some code

**Comment by tobarja** [comment]
Hey, you missed a comma in the `public static Task Delay` declaration, between provider and CancellationToken.

**Reply by linkdotnet** [reply]
Absolutely! Thanks for the hint - fixed.
